src/freenet/clients/http/wizardsteps/BROWSER_WARNING.java

```diff
@@ -104,6 +99,34 @@ public void getStep(HTTPRequest request, PageHelper helper) {
 		        new String[] { "submit", "next", NodeL10n.getBase().getString("Toadlet.next")});
 	}
 
+	public List<String> oldBrowserWarnings(
+			boolean incognito,
+			boolean isOldFirefox,
+			boolean showTabWarning) {
+		ArrayList<String> oldBrowserWarnings = new ArrayList<>();
+		if(isOldFirefox) {
+			oldBrowserWarnings.add(WizardL10n.l10n("browserWarningOldFirefox"));
+			if (showTabWarning) {
+				oldBrowserWarnings.add(WizardL10n.l10n("browserWarningFirefoxMightHaveClobberedTabs"));
+			} else if(!incognito) {
+				oldBrowserWarnings.add(WizardL10n.l10n("browserWarningOldFirefoxNewerHasPrivacyMode"));
+			}
+		}
+		return oldBrowserWarnings;
+	}
+
+	public String infoBoxHeaderText(boolean incognito, boolean isRelativelySafe) {
+		String infoBoxHeader;
+		if(incognito) {
+			infoBoxHeader = WizardL10n.l10n("browserWarningIncognitoShort");
+		} else if (isRelativelySafe) {
+			infoBoxHeader = WizardL10n.l10n("browserWarningShortRelativelySafe");
+		} else {
+			infoBoxHeader = WizardL10n.l10n("browserWarningShort");
+		}
+		return infoBoxHeader;
```

Comment:
I’m not a big fan of the one-return-per-method rule of thumb that was somehow created from the single-exit rule from assembly programming times (where it made sense!); just return the value as soon as you have it. It also makes the method easier to understand, IMO.

```
if (incognito) {
    return a;
}
if (isRelativelySafe) {
    return b;
}
return c;
```

Plus, the indentation after your `if`s is inconsistent. 🙂

Reply:
That’s the "let Intellij do the refactoring"-single return rule :smile:  , but you’re right.